qt_ui/windows/AirWingConfigurationDialog.py

```diff
@@ -273,25 +369,75 @@ def icon_for(aircraft: AircraftType) -> Optional[QIcon]:
 
 
 class AirWingConfigurationTab(QWidget):
-    def __init__(self, air_wing: AirWing, theater: ConflictTheater) -> None:
+    def __init__(self, coalition: Coalition, game: Game) -> None:
         super().__init__()
 
-        layout = QHBoxLayout()
+        layout = QGridLayout()
         self.setLayout(layout)
+        self.game = game
+        self.coalition = coalition
+
+        self.type_list = AircraftTypeList(coalition.air_wing)
+
+        layout.addWidget(self.type_list, 1, 1, 1, 2)
+
+        add_button = QPushButton("Add Squadron")
+        add_button.clicked.connect(lambda state: self.add_squadron())
+        layout.addWidget(add_button, 2, 1, 1, 1)
+
+        self.squadrons_panel = AircraftSquadronsPanel(coalition.air_wing, game.theater)
+        self.squadrons_panel.page_removed.connect(self.remove_aircraft_type)
+        layout.addLayout(self.squadrons_panel, 1, 3, 2, 1)
+
+        self.type_list.page_index_changed.connect(self.squadrons_panel.setCurrentIndex)
+
+    def remove_aircraft_type(self, aircraft_type: AircraftType):
+        for item in self.type_list.item_model.findItems(aircraft_type.name):
+            self.type_list.item_model.removeRow(item.row())
```

Comment:
Is this actually enough? Shouldn't squadron defs be returned to the air wing? I think if you pick your favorite hornet squadron, remove all the hornets here, then change your mind, you can't get that squadron back.

Reply:
Yes, this function just removes the UI representation of the aircraft_type in the left column list `type_list`. The squadron will be removed with the `remove squadron` button which triggers `remove_squadron`
https://github.com/dcs-liberation/dcs_liberation/blob/296000fdf732ae341acd11017210bd1b1f72cae8/qt_ui/windows/AirWingConfigurationDialog.py#L233-L241

This triggers this function:
https://github.com/dcs-liberation/dcs_liberation/blob/296000fdf732ae341acd11017210bd1b1f72cae8/qt_ui/windows/AirWingConfigurationDialog.py#L272-L274

Which will remove the whole `AircraftSquadronsPage` which represents the aircraft_type from UI Data Model.

The trick i made is in the apply function. I just clear the whole squadrons config with `self.air_wing.squadrons = {}` and the apply function just iterates over existing pages and theire squadrons. So removed pages -> removed aircraft_types

https://github.com/dcs-liberation/dcs_liberation/blob/296000fdf732ae341acd11017210bd1b1f72cae8/qt_ui/windows/AirWingConfigurationDialog.py#L320-L323

Just tested it again to be sure that removed aircrafts will not be added and its true :) Or do you see any problem here?

Reply:
No, I suspected I was missing something.